src/Fixer/Whitespace/BlankLineBeforeControlStatementFixer.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Whitespace;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\Fixer\WhitespacesAwareFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ * @author Andreas Möller <am@localheinz.com>
+ */
+final class BlankLineBeforeControlStatementFixer extends AbstractFixer implements ConfigurableFixerInterface, WhitespacesAwareFixerInterface
+{
+    /**
+     * @var array
+     */
+    private static $defaultConfiguration = array(
+        'return',
+    );
+
+    /**
+     * @var array
+     */
+    private static $tokenMap = array(
+        'break' => T_BREAK,
+        'continue' => T_CONTINUE,
+        'if' => T_IF,
+        'return' => T_RETURN,
+        'switch' => T_SWITCH,
+    );
+
+    /**
+     * @var array
+     */
+    private $fixTokenMap = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $configuration = self::$defaultConfiguration;
+        }
+
+        foreach ($configuration as $controlStatement) {
+            if (!is_string($controlStatement) || !array_key_exists($controlStatement, self::$tokenMap)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf(
+                    'Expected one of "%s", got "%s".',
+                    implode('", "', array_keys(self::$tokenMap)),
+                    is_object($controlStatement) ? get_class($controlStatement) : gettype($controlStatement)
+                ));
+            }
+
+            $this->fixTokenMap[$controlStatement] = self::$tokenMap[$controlStatement];
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
```

Comment:
please move after the getters (i.e. sort public the methods A-Z, normally I wouldn't care, but I'm re-ordering those everywhere atm)